### ISSUE TITLE
APIs inconsistent: some JSON, some XML, some both (and even some CSV)

### ISSUE THREAD

**Issue by maxrothman** [gocd/gocd#572]
Go's APIs currently have a range of output formats:

* **Pipeline**: XML (must be specified), plus custom verb endpoints
* **Agent**: JSON (not specified), plus custom verb endpoints (e.g. go/api/agents/:uuid/delete could be DELETE go/api/agents/:uuid)
* **Artifacts**: HTML/JSON (defaults to HTML)
* **Properties**: CSV only
* **Configuration**: XML (specified) for GETs, JSON (specified) for POSTs

In short, there's a mix of XML, JSON and CSV, some specified with a file extension, some not. Making this consistent would make it easier to write and iterate on API consumers.

**Comment by ciotlosm** [comment]
+1 for JSON

**Comment by zabil** [comment]
Refer #1100 for progress on this.

**Comment by arvindsv** [comment]
This has changed significantly over the last year and more. So, closing this. Let's open more specific issues for the ones that need to be changed.
